prov/efa/src/rdm/rxr_pkt_entry.c

```diff
@@ -74,13 +75,9 @@ struct rxr_pkt_entry *rxr_pkt_entry_alloc(struct rxr_ep *ep, struct rxr_pkt_pool
 		assert(pkt_entry->send);
 	}
 
-#ifdef ENABLE_EFA_POISONING
```

Comment:
The intention of memory poisoning is that when a new buffer is allocated, the buffer is poisoned, and if the code did not properly initialize some field in the buffer. The code will fail. Set the whole region to 0 defeats the purpose of memory poison.

Reply:
In that case, we should set the buffer to the poison value `0xDEADBEEF` instead of 0, right? If we remove the `memset` completely, then the buffer is uninitialized..

Reply:
The buffer was poisoned when they are released in `rxr_pkt_entry_release`.

But I agree poison during allocation make more sense.

Reply:
added a separate commit for that.